src/app.md

```diff
@@ -271,6 +272,51 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-file-skip-dot] #ex-file-skip-dot
+<a name="ex-file-skip-dot"></a>
+##  Recursively find all files skipping the ones starting with dot
+
+[`WalkDir::filter_entry`] and [`std::iter::Iterator::filter`] qctuqlly make iterators that return only the elements that pass the closure you give to them, the difference is that [`WalkDir::filter_entry`] is specialized to handle recursive directory traversal and it will also skip whole directories if the closure is not meet.
+
+[`WalkDir::depth`] is used to skip the level 0 which is the current directory `"."`.
+
+[![walkdir-badge]][walkdir] [![cat-filesystem-badge]][cat-filesystem]
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate walkdir;
+
+use walkdir::{DirEntry, WalkDir, WalkDirIterator};
+#
+# error_chain! {
+#     foreign_links {
+#         WalkDir(walkdir::Error);
+#         Io(std::io::Error);
+#         SystemTime(std::time::SystemTimeError);
```

Comment:
Well at the moment error-chain cannot infer if you do not use one of the error variants and assumes that you are planning use new one in the future. So the easiest way to solve it is to remove/comment out all the error variants and check if compiler starts to complain :wink:.

In this case no error variant is needed and we might as well drop error_chain usage here all together and switch `fn run` to `fn main`